lib/rules/no-duplicate-case.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview Rule to disallow a duplicate case label.
+ * @author Dieter Oberkofler
+ * @copyright 2015 Dieter Oberkofler. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    function isEqual(first, second) {
+        if (first.type === "Literal" && second.type === "Literal" && first.value === second.value) {
+            return true;
+        }
+
+        if (first.type === "Identifier" && second.type === "Identifier" && first.name === second.name) {
+            return true;
+        }
+
+        return false;
+    }
+
+    return {
+
+        "SwitchStatement": function(node) {
+
+            var noOfCases = node.cases.length,
+                i,
+                j;
+
+            // check all cases except the last one (-1)
+            for (i = 0; i < noOfCases - 1; i++) {
```

Comment:
What if the last `switch` is not `default`? It's better to check for `test` property of `null` to detect `default`, which you already do on the next line.